libraries/AP_Scripting/examples/gps_switch.lua

```diff
@@ -0,0 +1,60 @@
+-- Lua script to switch between GPS1 and GPS2 using an RC AUX switch
+
+-- To test in SITL:
+-- SIM_GPS2_ENABLE 1
+-- SIM_GPS2_TYPE 1
+-- GPS2_TYPE 1
+-- RC8_OPTION 300
+-- GPS_AUTO_SWITCH 0
+-- reboot
+-- param fetch GPS_PRIMARY
+-- rc 8 2000
+-- param fetch GPS_PRIMARY
+-- rc 8 1000
+-- param fetch GPS_PRIMARY
+
+
+-- The RCx_OPTION should be set to "300" (script function) for the desired channel
+local AUX_FUNCTION_NUM = 300
+
+-- Parameters
+local GPS_PRIMARY = Parameter("GPS_PRIMARY")
+local GPS_AUTO_SWITCH = Parameter("GPS_AUTO_SWITCH") -- should be 0 to allow manual switching
+
+-- Constants
+local GPS_FIRST = 0
+local GPS_SECOND = 1
+
+function update()
+   -- Read switch position (0=low, 1=mid, 2=high)
+   local sw_pos = rc:get_aux_cached(AUX_FUNCTION_NUM)
+   if sw_pos == nil then
+      gcs:send_text(1, "Waiting for RC input...")
```

Comment:
this doesn't mean it is waiting for R/C input, it just means the user hasn't hit the switch yet (for example AUX functions can be buttons on the GCS)
should just return with no msg

Reply:
in mavproxy use this: 
```
@alias add aux long DO_AUX_FUNCTION
```
and "aux 0", "aux 1", "aux 2"

Reply:
Done.